diff --git a/node_modules/react-native-mmkv/android/.project b/node_modules/react-native-mmkv/android/.project
new file mode 100644
index 0000000..ca2d399
--- /dev/null
+++ b/node_modules/react-native-mmkv/android/.project
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>react-native-mmkv</name>
+	<comment>Project react-native-mmkv created by Buildship.</comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>org.eclipse.buildship.core.gradleprojectbuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>org.eclipse.buildship.core.gradleprojectnature</nature>
+	</natures>
+	<filteredResources>
+		<filter>
+			<id>1743981374273</id>
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
+				<arguments>node_modules|\.git|__CREATED_BY_JAVA_LANGUAGE_SERVER__</arguments>
+			</matcher>
+		</filter>
+	</filteredResources>
+</projectDescription>
diff --git a/node_modules/react-native-mmkv/android/bin/CMakeLists.txt b/node_modules/react-native-mmkv/android/bin/CMakeLists.txt
new file mode 100644
index 0000000..9b7794c
--- /dev/null
+++ b/node_modules/react-native-mmkv/android/bin/CMakeLists.txt
@@ -0,0 +1,32 @@
+cmake_minimum_required(VERSION 3.9.0)
+project(ReactNativeMmkv)
+
+set(CMAKE_VERBOSE_MAKEFILE ON)
+set(CMAKE_CXX_STANDARD 17)
+
+# Compile sources
+add_library(
+        react-native-mmkv
+        SHARED
+        src/main/cpp/AndroidLogger.cpp
+        ../cpp/MmkvHostObject.cpp
+        ../cpp/NativeMmkvModule.cpp
+)
+
+# Add headers search paths
+target_include_directories(react-native-mmkv PUBLIC ../MMKV/Core)
+target_include_directories(react-native-mmkv PUBLIC ../cpp)
+
+# Add MMKV core dependency
+add_subdirectory(../MMKV/Core core)
+
+# Add android/log dependency
+find_library(log-lib log)
+
+target_link_libraries(
+        react-native-mmkv
+        core                        # <-- MMKV core
+        ${log-lib}                  # <-- Logcat logger
+        android                     # <-- Android JNI core
+        react_codegen_RNMmkvSpec    # <-- Generated Specs from CodeGen
+)
diff --git a/node_modules/react-native-mmkv/android/bin/build.gradle b/node_modules/react-native-mmkv/android/bin/build.gradle
new file mode 100644
index 0000000..934d4aa
--- /dev/null
+++ b/node_modules/react-native-mmkv/android/bin/build.gradle
@@ -0,0 +1,95 @@
+buildscript {
+  repositories {
+    google()
+    mavenCentral()
+  }
+
+  dependencies {
+    classpath "com.android.tools.build:gradle:7.2.1"
+  }
+}
+
+def isNewArchitectureEnabled() {
+  return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
+}
+
+apply plugin: "com.android.library"
+
+if (isNewArchitectureEnabled()) {
+  apply plugin: "com.facebook.react"
+}
+
+def getExtOrDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties["Mmkv_" + name]
+}
+
+def getExtOrIntegerDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties["Mmkv_" + name]).toInteger()
+}
+
+logger.warn("[react-native-mmkv] Thank you for using react-native-mmkv ❤️")
+logger.warn("[react-native-mmkv] If you enjoy using react-native-mmkv, please consider sponsoring this project: https://github.com/sponsors/mrousavy")
+
+android {
+  namespace "com.mrousavy.mmkv"
+
+  ndkVersion getExtOrDefault("ndkVersion")
+  compileSdkVersion getExtOrIntegerDefault("compileSdkVersion")
+
+  defaultConfig {
+    minSdkVersion getExtOrIntegerDefault("minSdkVersion")
+    targetSdkVersion getExtOrIntegerDefault("targetSdkVersion")
+    buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
+  }
+
+  buildFeatures {
+    buildConfig true
+  }
+
+  buildTypes {
+    release {
+      minifyEnabled false
+    }
+  }
+
+  lintOptions {
+    disable "GradleCompatible"
+  }
+
+  compileOptions {
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+  }
+
+  sourceSets {
+    main {
+      if (isNewArchitectureEnabled()) {
+          java.srcDirs += [
+            // This is needed to build Kotlin project with NewArch enabled
+            "${project.buildDir}/generated/source/codegen/java"
+          ]
+      }
+    }
+  }
+}
+
+repositories {
+  mavenCentral()
+  google()
+}
+
+
+dependencies {
+  // For < 0.71, this will be from the local maven repo
+  // For > 0.71, this will be replaced by `com.facebook.react:react-android:$version` by react gradle plugin
+  //noinspection GradleDynamicVersion
+  implementation "com.facebook.react:react-native:+"
+}
+
+if (isNewArchitectureEnabled()) {
+  react {
+    jsRootDir = file("../src/")
+    libraryName = "Mmkv"
+    codegenJavaPackageName = "com.mrousavy.mmkv"
+  }
+}
diff --git a/node_modules/react-native-mmkv/android/bin/gradle.properties b/node_modules/react-native-mmkv/android/bin/gradle.properties
new file mode 100644
index 0000000..5b9a8d9
--- /dev/null
+++ b/node_modules/react-native-mmkv/android/bin/gradle.properties
@@ -0,0 +1,5 @@
+Mmkv_kotlinVersion=1.9.24
+Mmkv_minSdkVersion=23
+Mmkv_targetSdkVersion=34
+Mmkv_compileSdkVersion=34
+Mmkv_ndkversion=26.1.10909125
diff --git a/node_modules/react-native-mmkv/android/bin/src/main/AndroidManifest.xml b/node_modules/react-native-mmkv/android/bin/src/main/AndroidManifest.xml
new file mode 100644
index 0000000..a2f47b6
--- /dev/null
+++ b/node_modules/react-native-mmkv/android/bin/src/main/AndroidManifest.xml
@@ -0,0 +1,2 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+</manifest>
diff --git a/node_modules/react-native-mmkv/android/bin/src/main/cpp/AndroidLogger.cpp b/node_modules/react-native-mmkv/android/bin/src/main/cpp/AndroidLogger.cpp
new file mode 100644
index 0000000..bd953b0
--- /dev/null
+++ b/node_modules/react-native-mmkv/android/bin/src/main/cpp/AndroidLogger.cpp
@@ -0,0 +1,16 @@
+//
+//  AndroidMmkvLogger.cpp
+//  react-native-mmkv
+//
+//  Created by Marc Rousavy on 05.03.24.
+//
+
+#include "MmkvLogger.h"
+#include <android/log.h>
+
+void MmkvLogger::log(const std::string& tag, const std::string& message) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-security"
+  __android_log_print(ANDROID_LOG_INFO, tag.c_str(), message.c_str());
+#pragma clang diagnostic pop
+}
diff --git a/node_modules/react-native-mmkv/android/bin/src/main/java/com/mrousavy/mmkv/MmkvPackage.class b/node_modules/react-native-mmkv/android/bin/src/main/java/com/mrousavy/mmkv/MmkvPackage.class
new file mode 100644
index 0000000..341ebbf
Binary files /dev/null and b/node_modules/react-native-mmkv/android/bin/src/main/java/com/mrousavy/mmkv/MmkvPackage.class differ
diff --git a/node_modules/react-native-mmkv/android/bin/src/main/java/com/mrousavy/mmkv/MmkvPlatformContextModule.class b/node_modules/react-native-mmkv/android/bin/src/main/java/com/mrousavy/mmkv/MmkvPlatformContextModule.class
new file mode 100644
index 0000000..87310cc
Binary files /dev/null and b/node_modules/react-native-mmkv/android/bin/src/main/java/com/mrousavy/mmkv/MmkvPlatformContextModule.class differ
diff --git a/node_modules/react-native-mmkv/react-native-mmkv.podspec b/node_modules/react-native-mmkv/react-native-mmkv.podspec
index be617c1..c495841 100644
--- a/node_modules/react-native-mmkv/react-native-mmkv.podspec
+++ b/node_modules/react-native-mmkv/react-native-mmkv.podspec
@@ -31,8 +31,37 @@ Pod::Spec.new do |s|
     "ios/**/*.{h,m,mm}",
     "cpp/**/*.{hpp,cpp,c,h}",
     # MMKV/Core
-    "MMKV/Core/**/*.{h,cpp,hpp,S}",
+    "MMKV/Core/**/*.{h,hpp,S}", # Header/template/assembly files
+    # Include only non-Win32 cpp files from MMKV/Core
+    "MMKV/Core/*.cpp", 
+    "MMKV/Core/aes/*.cpp",
+    "MMKV/Core/aes/openssl/*.cpp",
+    "MMKV/Core/crc32/*.cpp",
+    "MMKV/Core/crc32/zlib/*.cpp",
   ]
+  s.exclude_files = [
+    "MMKV/Core/*_Win32.cpp",
+    "MMKV/Core/**/*_Android.*",
+    "MMKV/Core/**/*_Linux.*",
+    "MMKV/Core/**/*_OSX.*", # Exclude specific OSX implementations if causing issues, keep otherwise
+    "MMKV/Core/**/aes/openssl/*armv4.S", # Example: Exclude specific assembly if needed
+  ]
+
+  # Add platform specific source files if needed
+  s.ios.source_files = [
+    "MMKV/Core/*_OSX.cpp", # Include OSX files only for iOS/macOS
+    "MMKV/Core/**/aes/openssl/*armx.S", # Include ARM assembly for iOS
+  ]
+  s.tvos.source_files = [
+    "MMKV/Core/*_OSX.cpp",
+    "MMKV/Core/**/aes/openssl/*armx.S",
+  ]
+  s.osx.source_files = [
+    "MMKV/Core/*_OSX.cpp",
+  ]
+  
+  # Add spec for vendored libraries if MMKV uses any (example)
+  # s.vendored_libraries = 'libs/libMMKV.a'
 
   install_modules_dependencies(s)
 end
